Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/Render/MarkdownRenderer.Blocks.cs

```diff
@@ -0,0 +1,432 @@
+                    paragraph.FontSize = Header4FontSize;
+                    paragraph.FontWeight = Header4FontWeight;
+                    paragraph.Foreground = Header4Foreground;
+                    break;
+
+                case 5:
+                    paragraph.Margin = Header5Margin;
+                    paragraph.FontSize = Header5FontSize;
+                    paragraph.FontWeight = Header5FontWeight;
+                    paragraph.Foreground = Header5Foreground;
+                    break;
+
+                case 6:
+                    paragraph.Margin = Header6Margin;
+                    paragraph.FontSize = Header6FontSize;
+                    paragraph.FontWeight = Header6FontWeight;
+                    paragraph.Foreground = Header6Foreground;
+
+                    var underline = new Underline();
+                    childInlines = underline.Inlines;
+                    paragraph.Inlines.Add(underline);
+                    break;
+            }
+
+            // Render the children into the para inline.
+            var childContext = new InlineRenderContext(childInlines, context)
+            {
+                Parent = paragraph,
+                TrimLeadingWhitespace = true
+            };
+
+            RenderInlineChildren(element.Inlines, childContext);
+
+            // Add it to the blocks
+            textBlock.Blocks.Add(paragraph);
+        }
+
+        /// <summary>
+        /// Renders a list element.
+        /// </summary>
+        protected override void RenderListElement(ListBlock element, IRenderContext context)
+        {
+            var localContext = context as UIElementCollectionRenderContext;
+            if (localContext == null)
+            {
+                throw new RenderContextIncorrectException();
+            }
+
+            var blockUIElementCollection = localContext.BlockUIElementCollection;
+
+            // Create a grid with two columns.
+            Grid grid = new Grid
+            {
+                Margin = ListMargin
+            };
+
+            // The first column for the bullet (or number) and the second for the text.
+            grid.ColumnDefinitions.Add(new ColumnDefinition() { Width = new GridLength(ListGutterWidth) });
+            grid.ColumnDefinitions.Add(new ColumnDefinition() { Width = new GridLength(1, GridUnitType.Star) });
+
+            for (int rowIndex = 0; rowIndex < element.Items.Count; rowIndex++)
+            {
+                var listItem = element.Items[rowIndex];
+
+                // Add a row definition.
+                grid.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
+
+                // Add the bullet or number.
+                var bullet = CreateTextBlock(localContext);
+                bullet.Margin = ParagraphMargin;
+                switch (element.Style)
+                {
+                    case ListStyle.Bulleted:
+                        bullet.Text = "•";
```

Comment:
Should we use escape sequence vs. explicit unicode character?  Should what this character is be configurable?

Reply:
This was an existing implementation.